bin/deployables

```diff
@@ -83,600 +99,829 @@ region=${DEPLOY_AWS_REGION}
 output=json
 EOH
 
-    if [[ $? != 0 ]] ; then
-        echo "Error updating aws config, DEPLOY_AWS_CONFIG=${DEPLOY_AWS_CONFIG}"
-        exit 1
-    fi
+  if [[ $? != 0 ]] ; then
+    echo "Error updating aws config, DEPLOY_AWS_CONFIG=${DEPLOY_AWS_CONFIG}"
+    exit 1
+  fi
+
+  cat $( eval echo -n $DEPLOY_AWS_CONFIG )
 
-    aws_configure_done=1
+  aws_configure_done=1
 }
 
-function ecr_login () {
+function aws_test_profile {
+  aws_configure
+  aws --profile=${DEPLOY_AWS_PROFILE} s3 ls
+}
 
-    echo
-    echo Logging in to ECR
-    echo DEPLOY_AWS_PROFILE: ${DEPLOY_AWS_PROFILE}
-    echo DEPLOY_AWS_ACCOUNT: ${DEPLOY_AWS_ACCOUNT}
-    echo DEPLOY_ECR_ACCOUNT: ${DEPLOY_ECR_ACCOUNT}
-    echo
+function aws_run () {
+  aws_configure
+  aws --profile=${DEPLOY_AWS_PROFILE} ${@:1}
+}
 
-	# check environment variables
-	check_environment
+function ecr_login () {
+  echo
+  echo Logging in to ECR
+  echo DEPLOY_AWS_PROFILE: ${DEPLOY_AWS_PROFILE}
+  echo DEPLOY_AWS_ACCOUNT: ${DEPLOY_AWS_ACCOUNT}
+  echo DEPLOY_ECR_ACCOUNT: ${DEPLOY_ECR_ACCOUNT}
+  echo
+
+  # check environment variables
+  check_environment
+
+  # Newer versions of docker don't work well with the default login args to aws-cli
+  email=""
+  if [[ $(docker --version | grep -i version) =~ 1(7|8)\. ]]; then
```

Comment:
So this is because aws versions and docker versions don't get along. Some versions of the AWS cli don't get along with certain versions of docker. This was an issue with Circle 1.0. Now that we're on Circle 2.0, we chould actually just always use this check, because we choose the AWS cli version, and docker is always a newish version.